forms/utils.py

```diff
@@ -330,3 +348,121 @@ def filter_queryset(self, request, queryset, view):
+
+
+def generate_and_queue_answer_emails(input_data: AnswerInputData) -> None:
+    from forms.models import Answer
+    from plotsearch.utils import build_pdf_context
+
+    email_messages: List[EmailMessage] = []
+    answer_id = input_data.get("answer_id")
+    answer_type = input_data.get("answer_type")
+
+    try:
+        answer = Answer.objects.get(id=answer_id)
+    except Answer.DoesNotExist:
+        logger.error(
+            f"Answer with id {answer_id} does not exist, unable to generate emails."
+        )
+        return
+
+    context = {}
+    attachments = []
+
+    if answer_type == AnswerType.TARGET_STATUS:
+        target_statuses = getattr(answer, "statuses", None)
+        target_status_identifiers = ", ".join(
+            target_statuses.values_list("application_identifier", flat=True)
+        )
+        from_email = settings.FROM_EMAIL_PLOT_SEARCH
+        email_subject = _(f"Copy of plot application(s) {target_status_identifiers}")
+        email_body = render_to_string("target_status/email_detail.txt", context)
+
+        for target_status in target_statuses:
+            context["object"] = target_status
+            context = build_pdf_context(context)
+            email_pdf = generate_pdf("target_status/detail.html", context=context)
+            attachment_filename = f"{area_search.application_identifier}.pdf"
+            attachments.append([attachment_filename, email_pdf, "application/pdf"])
+
+    if answer_type == AnswerType.AREA_SEARCH:
+        area_search = getattr(answer, "area_search", None)
+        context["object"] = area_search
+        context = build_pdf_context(context)
+        from_email = settings.FROM_EMAIL_AREA_SEARCH
+        email_subject = _(f"Copy of area rental application {area_search.identifier}")
+        email_body = render_to_string("area_search/email_detail.txt", context)
+        email_pdf: BytesIO = generate_pdf("area_search/detail.html", context=context)
+        attachment_filename = f"{getattr(area_search, 'identifier', email_subject)}.pdf"
+        attachments.append([attachment_filename, email_pdf, "application/pdf"])
+
+    for applicant in context.get("applicants", []):
+        try:
+            email_address = (
+                applicant["entry_section"]
+                .entries.filter(field__identifier="sahkoposti")
+                .first()
+                .value
+            )
+        except AttributeError:
+            continue
+
+        email_messages.append(
+            EmailMessage(
+                from_email=from_email or settings.MVJ_EMAIL_FROM,
+                to=[email_address],
+                subject=email_subject,
+                body=email_body,
+                attachments=attachments,
+            )
+        )
+
+    for email_message in email_messages:
+        result = {"answer_id": answer_id, "email_message": email_message}
+        async_task(
+            send_answer_email, input_data=result,
+        )
+
+    return
+
```

Comment:
Does this return here do anything?

Reply:
It does not, which is indicated in the return type `-> None`